common/deftype/deftype.cpp

```diff
@@ -1623,12 +1651,23 @@ extern DEFTYPE_API ITypeInfo *makeIntType(int len, bool isSigned)
     if (len <= 0 || len > 8)
         return NULL;
 
-    CriticalBlock procedure(*typeCS);
-    CIntTypeInfo *ret = itt[isSigned][len-1];
-    if (ret==NULL)
-        ret = itt[isSigned][len-1] = new CIntTypeInfo(len, isSigned);
-    ::Link(ret);
-    return ret;
+    std::atomic<CIntTypeInfo*>& singleton = itt[isSigned][len-1];
+    CIntTypeInfo* val = singleton.load();
+    if (val == nullptr)
+    {
+        CIntTypeInfo* newVal = new CIntTypeInfo(len, isSigned);
+        if (!singleton.compare_exchange_strong(val, newVal))
```

Comment:
readability: Although this code is correct I had to read it several times to check that val was correctly set up on all the paths.
Add a comment should to the true branch, or change to always assign val to singleton.load() after the if.